lib/honeybadger/agent.rb

```diff
@@ -125,6 +125,17 @@ def notify(exception_or_opts, opts = {})
 
       notice = Notice.new(config, opts)
 
+      if defined?(::Rails) && ::Rails.application.config.exceptions_app.is_a?(::ActionDispatch::Routing::RouteSet)
```

Comment:
It might be cleaner if we configured a [`before_notify` handler](https://docs.honeybadger.io/ruby/gem-reference/configuration.html#changing-notice-data) in [the Rails plugin](https://github.com/honeybadger-io/honeybadger-ruby/blob/master/lib/honeybadger/plugins/rails.rb) vs. adding the framework-specific code here in `Agent`. Our [integration Ruby guide](https://docs.honeybadger.io/ruby/gem-reference/integration.html) explains how the plugin system works. Do you think that could work?

Reply:
Followup: oops, somehow I missed the "alternative solution" examples when I first read the PR! :)

I think we might be able to use "Alternative Solution 1", but add the config automatically in the Rails plugin like I mentioned. I would be good with that. We could also optionally add a config flag to override the default behavior (the `config` object is available in all plugin callbacks).

The other option is to just recommend that users add "Alternative Solution 1" to their config manually if they want this behavior. We could add a guide on it to our docs site. I'm OK with that approach too, since very few people have asked for this. That said, anywhere that we can reliably enhance the default experience without causing integration issues is a plus, imo.

Reply:
oh yeah, this is similar to my idea of alternative solution `#2`, as I'm new to the agent, I was uncomfortable adding the code on the agent, but was unsure where to add, thank you for pointing me that.

is this valid to enabled via config flag, or be _enabled by default™_ on the plugin?

I'm gonna scratch some solution here then I push here 😄 

thank you

Reply:
I had a _bad time_ with this "already initialized" hahah, but I managed to test with just one rails instance! now I'm gonna check on the build if there is any of this template error, I think all of them are gonna be ok

the code now is on the rails plugin, easier to isolate / test... if everything is good (you think the code is on the right direction and build is green), then I can add a config _flag_ to enable/disable this behavior...